### PR TITLE
Add support for more than 1 level of dialect inclusion

### PR DESCRIPTION
**Pull request by hamishwillee**
The toolchain only supports including 1 level of dialect `include` nesting. This PR allows include files to be nested up to 5 levels deep (but will stop earlier if no new files are detected).

Current set of dialect files are unchanged by this.

### CODE REVIEW

File: generator/mavgen.py [hunk 1]
```diff
@@ -101,31 +142,12 @@ def mavgen_validate(xmlfile):
...
-            # include message lengths and CRCs too
-            x.message_crcs.update(xml[-1].message_crcs)
-            x.message_lengths.update(xml[-1].message_lengths)
-            x.message_min_lengths.update(xml[-1].message_min_lengths)
-            x.message_flags.update(xml[-1].message_flags)
-            x.message_target_system_ofs.update(xml[-1].message_target_system_ofs)
-            x.message_target_component_ofs.update(xml[-1].message_target_component_ofs)
-            x.message_names.update(xml[-1].message_names)
-            x.largest_payload = max(x.largest_payload, xml[-1].largest_payload)
+    for i in range(5):
```
Comment (hamishwillee) [inline generator/mavgen.py]: FYI Max 5 levels of nesting hard coded. We could put as a value up top of file, but I figure this is most we'll need for foreseeable future.

**Comment by hamishwillee** [comment]
@peterbarker When you have chance to review ...

**Comment by hamishwillee** [comment]
Can this please get reviewed?

**Comment by hamishwillee** [comment]
Ping.

**Comment by hamishwillee** [comment]
Great. @amilcarlucas appreciate you chasing this in the dev call.

**Comment by hamishwillee** [comment]
Hi @amilcarlucas - any luck on getting review on this in devcall?

**Comment by amilcarlucas** [comment]
@hamishwillee sorry did not attend the meeting this week :(

**Comment by hamishwillee** [comment]
No worries @amilcarlucas . This is in no way urgent.

File: generator/mavgen.py [hunk 2]
```diff
@@ -61,6 +64,39 @@ def mavgen(opts, args):
             print("WARNING: Unable to load XML validator libraries. XML validation will not be performed", file=sys.stderr)
             opts.validate = False
 
+    def expand_includes():
+        """Expand includes in current list of all files, ignoring those already parsed."""
+        for x in xml[:]:
+            for i in x.include:
+                fname = os.path.join(os.path.dirname(x.filename), i)
+
+                # Only parse new include files
```
Comment (hamishwillee) [inline generator/mavgen.py]: > please make it skip an xml if its already included

@tridge Already does - here.

**Comment by hamishwillee** [comment]
@tridge 

> Also, please test that this produces no change in headers for ardupilot xml

I retested that for Python and C for all dialects. No changes, because only 1 level of inclusion has been used.

**Comment by amilcarlucas** [comment]
Thanks

**Comment by hamishwillee** [comment]
Thanks @amilcarlucas . Now we just have to find a use for it :-) I was originally hoping to use this to include minimal.xml into common.xml - but all a bit scary.

**Pull request merged**